File: doconnect_frontend_final/src/services/answerService.js
/** 

 * Answer Service 

 * All answer-related API calls 

 */ 

import axiosInstance from '../api/axiosConfig'; 

 

const answerService = { 

 

  getAnswersByQuestion: async (questionId) => { 

    const res = await axiosInstance 

                .get(`/answers/${questionId}`); 

    return res.data; 

  }, 

 

  postAnswer: async (data) => { 

    const res = await axiosInstance 

                    .post('/answers', data); 

    return res.data; 

  }, 

 

  updateAnswer: async (id, data) => { 

    const res = await axiosInstance 

                  .put(`/answers/${id}`, data); 

    return res.data; 

  }, 

 

  deleteAnswer: async (id) => { 

    const res = await axiosInstance 

                    .delete(`/answers/${id}`); 

    return res.data; 

  }, 

 

  acceptAnswer: async (id) => { 

    const res = await axiosInstance 

              .put(`/answers/${id}/accept`); 

    return res.data; 

  }, 

 

  voteAnswer: async (id, type) => { 

    const res = await axiosInstance 

        .put(`/answers/${id}/vote?type=${type}`); 

    return res.data; 

  }, 

}; 

 

export default answerService; 

File: DoConnect_Final/src/main/java/com/doconnect/service/interfaces/AnswerService.java
package com.doconnect.service.interfaces;

import java.util.List;

import com.doconnect.dto.answer.AnswerRequestDTO;
import com.doconnect.dto.answer.AnswerResponseDTO;

public interface AnswerService { 

    AnswerResponseDTO postAnswer( 

            AnswerRequestDTO dto, String email); 

    List<AnswerResponseDTO> getAnswersByQuestion( 

            Long questionId); 

    AnswerResponseDTO updateAnswer( 

            Long id, 

            AnswerRequestDTO dto, 

            String email); 

    void deleteAnswer(Long id, String email); 

    AnswerResponseDTO acceptAnswer( 

            Long id, String email); 

    AnswerResponseDTO voteAnswer( 

            Long id, String type); 

}
